SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/PartitionManager.cpp

```diff
@@ -216,102 +221,50 @@ struct FPartitionManager::Impl
 			}
 		}
 
-		for (const EntityDelta& Delta : WorkerView.GetViewDelta().EntityDeltas)
+		WorkersDispatcher.Advance();
+		SystemWorkersDispatcher.Advance();
+
+		TSet<Worker_EntityId_Key> WorkersToInspect = WorkersData.GetModifiedEntities();
+		const TSet<Worker_EntityId_Key>& SystemWorkerModified = SystemWorkersData.GetModifiedEntities();
+		if (SystemWorkerModified.Num() != 0)
 		{
-			switch (Delta.Type)
-			{
-			case EntityDelta::UPDATE:
+			for (const auto& Entry : WorkersData.GetObjects())
 			{
-				ServerWorker& ServerWorkerData = WorkersData.FindChecked(Delta.EntityId);
-				bool bWasReady = ServerWorkerData.bReadyToBeginPlay;
-				for (const auto& CompleteUpdate : Delta.ComponentsRefreshed)
+				if (SystemWorkerModified.Contains(Entry.Value.SystemEntityId))
 				{
-					if (CompleteUpdate.ComponentId == SpatialConstants::SERVER_WORKER_COMPONENT_ID)
-					{
-						ServerWorkerData = ServerWorker(CompleteUpdate.Data);
-					}
+					WorkersToInspect.Add(Entry.Key);
 				}
-				for (const auto& Update : Delta.ComponentUpdates)
-				{
-					if (Update.ComponentId == SpatialConstants::SERVER_WORKER_COMPONENT_ID)
-					{
-						ServerWorkerData.ApplyComponentUpdate(Update.Update);
-					}
-				}
-				if (!bWasReady && ServerWorkerData.bReadyToBeginPlay)
-				{
-					WorkerConnected(Delta.EntityId);
-				}
-			}
-			break;
-			case EntityDelta::ADD:
-			{
-				const EntityViewElement& WorkerDesc = WorkerView.GetView().FindChecked(Delta.EntityId);
-				for (const auto& Component : WorkerDesc.Components)
-				{
-					if (Component.GetComponentId() == SpatialConstants::SERVER_WORKER_COMPONENT_ID)
-					{
-						ServerWorker& ServerWorkerData = WorkersData.Add(Delta.EntityId, ServerWorker(Component.GetUnderlying()));
-						if (ServerWorkerData.bReadyToBeginPlay)
-						{
-							WorkerConnected(Delta.EntityId);
-						}
-					}
-				}
-			}
-			break;
-			case EntityDelta::REMOVE:
-			{
-				WorkersData.Remove(Delta.EntityId);
-			}
-			case EntityDelta::TEMPORARILY_REMOVED:
-			{
-			}
-			break;
-			default:
-				break;
 			}
 		}
 
-		for (const EntityDelta& Delta : SystemWorkerView.GetViewDelta().EntityDeltas)
+		for (Worker_EntityId UpdatedWorker : WorkersToInspect)
 		{
-			switch (Delta.Type)
+			const ServerWorker& ServerWorkerData = WorkersData.GetObjects().FindChecked(UpdatedWorker);
+			if (ServerWorkerData.bReadyToBeginPlay)
 			{
-			case EntityDelta::ADD:
-			{
-				const EntityViewElement& WorkerDesc = SystemWorkerView.GetView().FindChecked(Delta.EntityId);
-				for (const auto& Component : WorkerDesc.Components)
+				bool bIsAlreadyConnected = false;
+				for (FLBWorkerHandle Connected : ConnectedWorkers)
+				{
+					bIsAlreadyConnected |= Connected->State->ServerWorkerId == UpdatedWorker;
+				}
```

Comment:
Is the loop essentially a `ContainsByPredicate` (which we don't have on a TSet)? If so, maybe consider this implementation to reduce confusion.
P.S. Requires an additional include
```suggestion
				const LBWorkerHandle* ExistingConnectedWorkerPtr = Algo::FindByPredicate(ConnectedWorkers, [UpdatedWorker](const LBWorkerHandle ConnectedWorker)
				{
					return ConnectedWorker->State->ServerWorkerId == UpdatedWorker
				});
				const bool bIsAlreadyConnected = ExistingConnectedWorkerPtr != nullptr;
```